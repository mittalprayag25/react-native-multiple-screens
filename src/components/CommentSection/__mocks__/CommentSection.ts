import {createElement} from 'react';

jest.mock('..', () => createElement('CommentSection'));
