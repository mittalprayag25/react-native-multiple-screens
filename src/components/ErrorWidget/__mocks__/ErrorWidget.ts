import {createElement} from 'react';

jest.mock('..', () => createElement('ErrorWidget'));
