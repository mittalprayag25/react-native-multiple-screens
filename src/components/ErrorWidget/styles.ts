import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  margins: {
    marginVertical: 20,
    alignSelf: 'center',
    fontSize: 16,
  },
  container: {
    backgroundColor: 'white',
    padding: 10,
  },
  btn: {},
});

export default styles;
