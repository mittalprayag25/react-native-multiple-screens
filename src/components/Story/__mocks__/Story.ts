import {createElement} from 'react';

jest.mock('..', () => createElement('Story'));
