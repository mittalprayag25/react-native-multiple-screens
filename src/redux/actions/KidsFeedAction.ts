export default {
  REQUEST: (index: number) => index,
  SUCCESS: (payload: any) => payload,
  FAILURE: (payload: any) => payload,
};
