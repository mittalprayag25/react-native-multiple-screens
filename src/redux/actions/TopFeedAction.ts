export default {
  REQUEST: () => null,
  RESET: () => null,
  SUCCESS: (payload: any) => payload,
  FAILURE: (payload: any) => payload,
};
