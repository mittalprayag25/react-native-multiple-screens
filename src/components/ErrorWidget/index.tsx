import React, {memo} from 'react';
import {View, Text, Button} from 'react-native';
import {useDispatch} from 'react-redux';

import styles from './styles';

const ErrorWidget = ({
  label = 'Error occured',
  action = null,
}: any): JSX.Element => {
  const dispatch = useDispatch();
  const tryAgain = () => {
    if (action) {
      dispatch(action);
    }
  };

  return (
    <View style={styles.container}>
      <Text style={styles.margins}>{label}</Text>
      <Text style={styles.margins}>
        This may be due to technical difficulty.
      </Text>
      <Button testID="tryAgain" onPress={() => tryAgain()} title="Try again" />
    </View>
  );
};

export default memo(ErrorWidget);
