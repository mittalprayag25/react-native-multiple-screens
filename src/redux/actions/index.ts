import {createActions} from 'redux-actions';
import TopFeedAction from './TopFeedAction';
import FeedAction from './FeedAction';
import KidsFeedAction from './KidsFeedAction';

const actions: any = createActions({
  GET_TOP_FEED: TopFeedAction,
  GET_FEED: FeedAction,
  GET_KIDS_FEED: KidsFeedAction,
});

export default actions;
