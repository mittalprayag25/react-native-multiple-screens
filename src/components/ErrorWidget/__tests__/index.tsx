import {act, fireEvent, render} from '@testing-library/react-native';
import React from 'react';
import {Provider} from 'react-redux';

import {createMockStore} from '../../../__mocks__/index';
import actions from '../../../redux/actions';
import ErrorWidget from '..';

describe('ErrorWidget ', () => {
  const props = {
    label: 'label',
    action: actions.getTopFeed.request(),
  };

  it('should match snapshot', () => {
    const tree = render(
      <Provider store={createMockStore()}>
        <ErrorWidget {...props} />
      </Provider>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('Test Try again click', () => {
    const newMockStore = createMockStore();
    newMockStore.dispatch = jest.fn();
    const props = {
      label: 'label',
      action: actions.getTopFeed.request(),
    };
    const {getByTestId} = render(
      <Provider store={newMockStore}>
        <ErrorWidget {...props} />
      </Provider>,
    );
    const tryAgainLink = getByTestId('tryAgain');
    act(() => {
      fireEvent(tryAgainLink, 'press');
    });
    expect(newMockStore.dispatch).toHaveBeenCalledTimes(1);
    expect(newMockStore.dispatch).toHaveBeenCalledWith(
      actions.getTopFeed.request(),
    );
  });

  it('Test Try again click with null action', () => {
    const newMockStore = createMockStore();
    newMockStore.dispatch = jest.fn();
    const props = {
      label: 'label',
    };
    const {getByTestId} = render(
      <Provider store={newMockStore}>
        <ErrorWidget {...props} />
      </Provider>,
    );
    const tryAgainLink = getByTestId('tryAgain');
    act(() => {
      fireEvent(tryAgainLink, 'press');
    });
    expect(newMockStore.dispatch).not.toHaveBeenCalledTimes(1);
  });
});
