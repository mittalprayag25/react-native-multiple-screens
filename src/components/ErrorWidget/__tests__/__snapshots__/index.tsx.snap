// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorWidget  should match snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "padding": 10,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "fontSize": 16,
        "marginVertical": 20,
      }
    }
  >
    label
  </Text>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "fontSize": 16,
        "marginVertical": 20,
      }
    }
  >
    This may be due to technical difficulty.
  </Text>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="tryAgain"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Try again
      </Text>
    </View>
  </View>
</View>
`;
